include/mbedtls/config.h

```diff
@@ -2798,7 +2798,26 @@
 //#define MBEDTLS_SSL_CACHE_DEFAULT_MAX_ENTRIES      50 /**< Maximum entries in cache */
 
 /* SSL options */
-//#define MBEDTLS_SSL_MAX_CONTENT_LEN             16384 /**< Maxium fragment length in bytes, determines the size of each of the two internal I/O buffers */
+//#define MBEDTLS_SSL_MAX_CONTENT_LEN             16384 /**< Maximum fragment length in bytes, determines the size of each of the two internal I/O buffers */
+
+//#define MBEDTLS_SSL_IN_CONTENT_LEN              16384 /**< Optionally override MBEDTLS_SSL_MAX_CONTENT_LEN for the inward TLS buffer only. */
+
+/**
+ * It is possible to save RAM by setting a smaller outward buffer using
+ * MBEDTLS_SSL_OUT_CONTENT_LEN, while keeping the default inward 16384 byte
+ * buffer to conform to the TLS specification.
+ *
+ * The minimum required outgoing buffer size is determined by the handshake
+ * protocol's usage. 4096 bytes is usually adequate.
+ *
+ * (Using the RFC 6066 max_fragment_length extension is a better alternative to
```

Comment:
This is basically advice on how we could implement the buffer size differently. It should be raised as an issue, not present in the source code as a comment.

Reply:
I think there's some useful advice for the user of mbedTLS here, if their question is "how can I save RAM used by buffers?"

* *If* max_fragment_length is available at their peer, setting this and shrinking the overall buffer size gives the lowest overall RAM usage.
* Otherwise, reducing the outgoing buffer size is the only best option.

Maybe this can be reworded in a more user-centric way?